source/deimos/openssl/opensslv.d

```diff
@@ -28,21 +120,26 @@ import deimos.openssl._d_util;
  */
 
 /* Version macros for compile-time API version detection */
-enum
-{
-    OPENSSL_VERSION_MAJOR   = 1,
-    OPENSSL_VERSION_MINOR   = 1,
-    OPENSSL_VERSION_PATCH   = 0,
-    OPENSSL_VERSION_BUILD   = 'h' - '`'
-}
+deprecated("Use `OpenSSLVersion.Major` instead, which has a correct value")
+enum OPENSSL_VERSION_MAJOR   = 1;
+
+deprecated("Use `OpenSSLVersion.Minor` instead, which has a correct value")
+enum OPENSSL_VERSION_MINOR   = 1;
+
+deprecated("Use `OpenSSLVersion.Patch` instead, which has a correct value")
+enum OPENSSL_VERSION_PATCH   = 0;
+
+deprecated("Use `OpenSSLVersion.Build` instead, which has a correct value")
+enum OPENSSL_VERSION_BUILD   = 'h' - '`';
 
 int OPENSSL_MAKE_VERSION(int major, int minor, int patch, int build)
 {
     return (major << 28) | (minor << 20) | (patch << 12) | (build << 4) | 0xf;
 }
 
 enum OPENSSL_VERSION_NUMBER =
-    OPENSSL_MAKE_VERSION(OPENSSL_VERSION_MAJOR, OPENSSL_VERSION_MINOR, OPENSSL_VERSION_PATCH, OPENSSL_VERSION_BUILD);
+    OPENSSL_MAKE_VERSION(OpenSSLVersion.Major, OpenSSLVersion.Minor,
+                         OpenSSLVersion.Patch, OpenSSLVersion.Build);
```

Comment:
It seems especially odd that this will be kept up to date, but not the individual constants. If those were to be changed to point to the struct template, then this change becomes unnecessary.

Reply:
That's a good point. I decided to update this because OpenSSL has this constant has a way to version code. With our D bindings, I don't think that this is really useful TBH.

Reply:
I don't understand what you mean by "OpenSSL has this constant has a way to version code".

Reply:
as a way*

I meant that those were designed to work with C code and a preprocessor. Macros only make sense in a binding when they are used to alias an API. A better way to do what `OPENSSL_VERSION_NUMBER` does in D would be to implement an `opCmp` so that you could compare two versions.

But anyway, as it's the only point of contention, and I don't use those defs myself, I'll just make them alias the struct fields.